File: Python/Queue/BJAlgo_10845.py
from sys import stdin
que = []

for _ in range(int(stdin.readline())):
    comm = stdin.readline().split()

    if comm[0] == 'push':
        que.append(int(comm[1]))
    elif comm[0] == 'pop':
        if que:
            print(que.pop(0))
        else:
            print(-1)
    elif comm[0] == 'size':
        print(len(que))
    elif comm[0] == 'empty':
        print(1-int(bool(que)))
    elif comm[0] == 'front':
        if que:
            print(que[0])
        else:
            print(-1)
    elif comm[0] == 'back':
        if que:
            print(que[-1])
        else:
            print(-1)

File: Java/Queue/BJAlgo_10845.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;

public class BJAlgo_10845 {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int count = Integer.parseInt(br.readLine());
        List que = new ArrayList<>();

        for(int i = 0; i < count; i++) {
            String[] comm = br.readLine().split(" ");

            switch (comm[0]){
                case "push":
                    que.add(Integer.parseInt(comm[1]));
                    break;
                case "pop":
                    if (que.isEmpty()) {
                        System.out.println(-1);
                    } else {
                        int data = (int) que.get(0);
                        que.remove(que.get(0));
                        System.out.println(data);
                    }
                    break;
                case "size":
                    System.out.println(que.size());
                    break;
                case "empty":
                    if (que.isEmpty()) {
                        System.out.println(1);
                    } else {
                        System.out.println(0);
                    }
                    break;
                case "front":
                    if (que.isEmpty()) {
                        System.out.println(-1);
                    } else {
                        System.out.println(que.get(0));
                    }
                    break;
                case "back":
                    if (que.isEmpty()) {
                        System.out.println(-1);
                    } else {
                        System.out.println(que.get(que.size() - 1));
                    }
                    break;
            }
        }
    }
}
